src/pkg/k8s/tunnel.go

```diff
@@ -254,19 +255,20 @@ func (tunnel *Tunnel) getAttachablePodForService(ctx context.Context) (string, e
 	if err != nil {
 		return "", fmt.Errorf("unable to find the service: %w", err)
 	}
-	selectorLabelsOfPods := MakeLabels(service.Spec.Selector)
-
-	servicePods := tunnel.kube.WaitForPodsAndContainers(
-		ctx,
-		PodLookup{
-			Namespace: tunnel.namespace,
-			Selector:  selectorLabelsOfPods,
-		},
-		nil,
-	)
```

Comment:
Function is only used to list ready pods, probably not to wait for pods to run. Better instead to remove the wait logic.